TSMarkdownParser/TSMarkdownParser.m

```diff
@@ -160,10 +168,38 @@ + (instancetype)standardParser {
 static NSString *const TSMarkdownMonospaceRegex        = @"(`+)\\s*([\\s\\S]*?[^`])\\s*\\1(?!`)";
 
 - (void)addParagraphParsingWithFormattingBlock:(void(^)(NSMutableAttributedString *attributedString, NSRange range))formattingBlock {
-    self.paragraphParsingBlock = ^(NSMutableAttributedString *attributedString) {
-        
-        formattingBlock(attributedString, NSMakeRange(0, attributedString.length));
-    };
+    NSRegularExpression *paragraphParsing = [NSRegularExpression regularExpressionWithPattern:TSMarkdownParagraphRegex options:NSRegularExpressionDotMatchesLineSeparators error:nil];
+    
+    [self addInitialParsingRuleWithRegularExpression:paragraphParsing withBlock:^(NSTextCheckingResult *match, NSMutableAttributedString *attributedString) {
+        formattingBlock(attributedString, match.range);
+    }];
+}
+
+- (void)addEscapingParsing {
+    NSRegularExpression *escapingParsing = [NSRegularExpression regularExpressionWithPattern:TSMarkdownEscapingRegex options:NSRegularExpressionDotMatchesLineSeparators error:nil];
+    
+    [self addInitialParsingRuleWithRegularExpression:escapingParsing withBlock:^(NSTextCheckingResult *match, NSMutableAttributedString *attributedString) {
+        NSRange range = NSMakeRange(match.range.location+1, 1);
+        NSString *matchString = [attributedString attributedSubstringFromRange:range].string;
+        NSString *escapedString = [NSString stringWithFormat:@"%04x", [matchString characterAtIndex:0]];
+        [attributedString replaceCharactersInRange:range withString:escapedString];
+    }];
+    
+    NSRegularExpression *unescapingParsing = [NSRegularExpression regularExpressionWithPattern:TSMarkdownUnescapingRegex options:NSRegularExpressionDotMatchesLineSeparators error:nil];
+    
+    [self addFinalParsingRuleWithRegularExpression:unescapingParsing withBlock:^(NSTextCheckingResult *match, NSMutableAttributedString *attributedString) {
+        NSRange range = NSMakeRange(match.range.location+1, 4);
+        NSString *matchString = [attributedString attributedSubstringFromRange:range].string;
+        char byte_chars[5] = {'\0','\0','\0','\0','\0'};
+        byte_chars[0] = [matchString characterAtIndex:0];
+        byte_chars[1] = [matchString characterAtIndex:1];
+        byte_chars[2] = [matchString characterAtIndex:2];
+        byte_chars[3] = [matchString characterAtIndex:3];
+        unichar whole_char = strtol(byte_chars, NULL, 16);
```

Comment:
Maybe this works too: `unichar whole_char = atol([[matchString substringToIndex:3] UTF8String], NULL, 16);`
(needs testing)